### ISSUE TITLE
Enable WARN logging with DefaultHandlerExceptionResolver

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#2176]
Currently if an exception such as `MissingServletRequestParameterException` is thrown no stack-trace is logged. It would be useful to provide an option to add them.

Something like this in `EnableWebMvcConfiguration`:

```
		@Override
		protected void configureHandlerExceptionResolvers(
				List<HandlerExceptionResolver> exceptionResolvers) {
			addDefaultHandlerExceptionResolvers(exceptionResolvers);
			for (HandlerExceptionResolver resolver : exceptionResolvers) {
				if (resolver instanceof DefaultHandlerExceptionResolver) {
					((DefaultHandlerExceptionResolver) resolver)
							.setWarnLogCategory(DefaultHandlerExceptionResolver.class
									.getName());
				}
			}
		}
```

I'm not sure if this should be on or off by default.

**Comment by chrylis** [comment]
I'm not familiar enough with the internals of the MVC chain post-controller to say whether mine is a duplicate, but they're semantically related. This one seems to be dealing with resolving controller arguments, and mine is about exceptions thrown after the controller has returned an argument during JSON/XML serialization.

**Comment by snicoll** [comment]
@philwebb I gave that a try, you can find a proposal (see https://github.com/snicoll/spring-boot/commit/745671a99a4ec48e3aca055480bc74b0ef057fbe) 

Let's assume I have added a `@ResponseStatus` annotation on a custom exception, here's what I get for a _single_ call to the service that throws that exception:

```
2016-05-24 13:28:05.641  WARN 97219 --- [nio-8080-exec-1] .m.m.a.ExceptionHandlerExceptionResolver : Handler execution resulted in exception: com.example.DemoApplication$MyException
2016-05-24 13:28:05.643  WARN 97219 --- [nio-8080-exec-1] .w.s.m.a.ResponseStatusExceptionResolver : Handler execution resulted in exception: com.example.DemoApplication$MyException
```

Your proposal above does not work as MVC ships with 3 "default" resolvers and that default is the fallback. Spring MVC logs the exception even if the handler does not ultimately resolve the model (that's  what happens with `ExceptionHandlerExceptionResolver`).

Long story short:

* Enabling only `DefaultHandlerExceptionResolver` looks incomplete to me: there are two other implementations _before_ this one that have a chance to resolve the exception
* Enabling all 3 defaults can easily lead to duplicate warning log

Thoughts?

**Comment by celiovasconcelos** [comment]
I was forced to workaround like this:

```JAVA
@Slf4j //lombok
@ResponseStatus(HttpStatus.BAD_REQUEST)
public class PushLinkException extends RuntimeException {

  public PushLinkException(String msg) {
    super(msg);
    log.warn(msg, this);
  }

}
```
Otherwise, no way to see my custom exception in the log.

**Comment by bclozel** [comment]
[SPR-14392](https://jira.spring.io/browse/SPR-14392) has been fixed.

You should now get a single WARN log when the exception is actually resolved, like this:

    [.m.m.a.ExceptionHandlerExceptionResolver] - Resolved exception caused by Handler execution: java.lang.IllegalStateException

@snicoll I think you can now merge your proposal.

**Comment by fedotxxl** [comment]
Exception has been correctly processed by ExceptionHandler. What's the point to log it? What's the point to log it at WARN level? What's the point to enable it by default?

**Comment by TorosyanV** [comment]
When I have ControllerAdvice and have method anotated with ExceptionHandler and I'm doing redirect   ExceptionHandlerExceptionResolver logs with WARN level.
Why?
I already handled custom exception and logged it

**Comment by wilkinsona** [comment]
`ExceptionHandlerExceptionResolver` is part of Spring Framework. If you have a question about it, please ask on Stack Overflow. If you have a suggestion for an improvement, please open a [JIRA ticket](https://jira.spring.io/browse/SPR).

**Comment by MSMTFaro** [comment]
@TorosyanV I have same issue and having to fix by setting log level to ERROR. I wonder if you found a better way.
` org.springframework.web.servlet.mvc.method.annotation.ExceptionHandlerExceptionResolver: ERROR`
